pkg/identity/service.go

```diff
@@ -69,6 +154,14 @@ func Register(g *grpc.Server, s *Service) {
 
 // Certify validates identity and signs certificates.
 func (svc *Service) Certify(ctx context.Context, req *pb.CertifyRequest) (*pb.CertifyResponse, error) {
+	svc.issuerMutex.Lock()
```

Comment:
I am not sure about locking the entire function. Will it work if we create a `getIssuerSafe() *tls.Issuer` method and perform the lock there?

Reply:
It will work, but now that I look at it, I think we can use `RWMutex` and just do a `RLock/RUnlock` to eliminate any potential contention in this block that you might be worried about.

Reply:
Good call on `RWMutex`. Will it be used to lock just the `service.issuer` field or the `Certify` function?

Reply:
The lock does not lock a field, it locks a critical section. In this case there will be nothing waiting (being blocked) on read, despite the fact that the whole certify function is a critical section. Except for when there is concurrent write performed by the update function which in practice happens so infrequently (how often issuer certs are rotated?) and is such a quick operation that it should not matter in practice.  Does that make sense?